src/webob/request.py

```diff
@@ -414,7 +414,7 @@ def host_port(self):
         host = e.get('HTTP_HOST')
         if host is not None:
             if ':' in host:
```

Comment:
This check is not enough to determine to split it based on `:`. The IPv6 address may not have a port but still has `:` characters.